### PR TITLE
Fix cyclomatic complexity across P2P Library - Closes #3859

### LINKED ISSUE #3859: Fix cyclomatic complexity in P2P constructor
### Expected behavior
[Reference comment thread](https://github.com/LiskHQ/lisk-sdk/pull/3824#discussion_r297162071)

There should be no need to disable tslint warning for cyclomatic complexity.

### Actual behavior

In feature branch for p2p, we disable tslint warning.

### PR DESCRIPTION
**Pull request by diego-G**
### Review checklist

- [x] The PR resolves #3859
- [x] All new code is covered with unit tests
- [x] Relevant integration / functional tests are added
- [x] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [x] Documentation has been added/updated

### CODE REVIEW

File: elements/lisk-p2p/src/p2p.ts [hunk 1]
```diff
@@ -365,7 +364,36 @@ export class P2P extends EventEmitter {
...
+		this._populatorInterval = config.populatorInterval
+			? config.populatorInterval
+			: DEFAULT_POPULATOR_INTERVAL;
+
+		this._peerHandshakeCheck = config.peerHandshakeCheck
+			? config.peerHandshakeCheck
+			: validatePeerCompatibility;
+	}
+
+	private static _createPeerPoolConfig(config: P2PConfig): PeerPoolConfig {
```
Comment (ishantiw) [inline elements/lisk-p2p/src/p2p.ts]: why we are making it a `static` function if it's not a `public` function, it doesn't add any value to make it `static`

File: elements/lisk-p2p/src/p2p.ts [hunk 2]
```diff
@@ -154,7 +154,6 @@ export class P2P extends EventEmitter {
 	private readonly _handleInboundSocketError: (error: Error) => void;
 	private readonly _peerHandshakeCheck: P2PCheckPeerCompatibility;
 
-	// tslint:disable-next-line: cyclomatic-complexity
 	public constructor(config: P2PConfig) {
```
Comment (ishantiw) [inline elements/lisk-p2p/src/p2p.ts]: Can we organize all the initialization based on category? like it is for `constants.ts`, we can have `// Event handlers`, `Config`, `PeerPool`, `PeerBook`, etc

File: elements/lisk-p2p/src/p2p.ts (hunk 1 above)
Comment (shuse2) [reply elements/lisk-p2p/src/p2p.ts]: If it does not use the context, please use pure function

Comment (diego-G) [reply elements/lisk-p2p/src/p2p.ts]: Adressed.

**Pull request merged**